### ISSUE TITLE
Cannot parse comment lines betwen actual postings

### ISSUE THREAD

**Issue by Harry-Chen** [zhang-accounting/zhang#381]
Both zhang and beancount parser report error on input like:

```baencount
2022-06-22 * "Test"
 Expenses:A -0.0 CNY
; comment
 Assets:B
```

Errors are like:

```
Parse Error 
Path:  --> 4:2
  |
4 |  Assets:B
  |  ^---
  |
  = expected EOI or comment_prefix
```

I did not investigate into syntax definition yet.

**Comment by Kilerd** [comment]
I'm not sure if it's valid in original beancount parser. 
I will do an investigation. if it's valid, I will fix it.

**Comment by Harry-Chen** [comment]
I frequently use this (i.e. comment out a line anywhere) in beancount. So I think this is a supported syntax to write.

**Comment by Kilerd** [comment]
make sense, I will fix it in days

**Comment by Harry-Chen** [comment]
There are some more differences:

1. Detail lines belonging to one transaction are only required to be indented, but not necessarily consistent. And comments are allowed in any of these lines, as long as they are also indented.
2. No space is required in `posting_cost`, `posting_single_price`, and `posting_total_price` (i.e. you should replace with `space*`).

For example, such grammar is legal is bc:

```
2022-06-22 * "Test"
  Expenses:A -0.0CNY@10JPY
   ; test
 ; test2
 Assets:B 6 CNY @@ 0.2 USD
      ; test_many_spaces
; test_no_space
```

**Comment by Kilerd** [comment]
I've create the PR and started working on this issue.

And there are three testcases added, do you mind to have a look whether there are any cases I miss?
https://github.com/zhang-accounting/zhang/pull/383/files

**Comment by Kilerd** [comment]
the issue have been fixed in PR #383 and have been published in snapshot version.

It will be released to 0.2.0 version with some changes later.
